### PR TITLE
RavenDB-8054 RQL AutoComplete

### PR DESCRIPTION
**Pull request by fitzchak**

### CODE REVIEW

File: src/Raven.Studio/typescript/common/bindingHelpers/aceEditorBindingHandler.ts [hunk 1]
```diff
@@ -209,22 +211,18 @@ class aceEditorBindingHandler {
...
-                        langTools.addCompleter({ moduleId: "aceEditoBindingHandler", getCompletions: aceEditorBindingHandler.autoCompleteHub });
-                    }
-                }
-                else {
-                    langTools.addCompleter({ moduleId: "aceEditoBindingHandler", getCompletions: aceEditorBindingHandler.autoCompleteHub });
+                if (!aceEditorBindingHandler.customCompleters.find(x => x.editorType === typeName)) {
+                    aceEditorBindingHandler.customCompleters.push({editorType: typeName, completer: completer});
                 }
+                langTools.setCompleters([{
+                    moduleId: "aceEditoBindingHandler",
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/bindingHelpers/aceEditorBindingHandler.ts]: typo

File: src/Raven.Studio/typescript/common/queryCompleter.ts [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+
+import collectionsTracker = require("common/helpers/database/collectionsTracker");
+import getIndexTermsCommand = require("commands/database/index/getIndexTermsCommand");
+import getDocumentsMetadataByIDPrefixCommand = require("commands/database/documents/getDocumentsMetadataByIDPrefixCommand");
+import database = require("models/resources/database");
+import getIndexEntriesFieldsCommand = require("commands/database/index/getIndexEntriesFieldsCommand");
+import collection = require("models/database/documents/collection");
+import document = require("models/database/documents/document");
+
+interface AutoCompleteLastKeyword {
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/queryCompleter.ts]: we use lower case for interfaces as convention in studio.

File: src/Raven.Studio/typescript/common/queryCompleter.ts [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+
+                    this.getIndexFields(session)
+                        .done((indexFields) => {
+                            if (!indexFields.find(x => x === currentField)) {
+                                return;
+                            }
+
+                            let currentValue: string = "";
+
+                            /*currentValue = currentToken.value.trim();
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/queryCompleter.ts]: remove?

File: src/Raven.Studio/typescript/common/bindingHelpers/aceEditorBindingHandler.ts (hunk 1 above)
Comment (fitzchak) [reply src/Raven.Studio/typescript/common/bindingHelpers/aceEditorBindingHandler.ts]: Fixed

File: src/Raven.Studio/typescript/common/queryCompleter.ts (hunk 2 above)
Comment (fitzchak) [reply src/Raven.Studio/typescript/common/queryCompleter.ts]: Right. Fixed.

File: src/Raven.Studio/typescript/common/queryCompleter.ts (hunk 3 above)
Comment (fitzchak) [reply src/Raven.Studio/typescript/common/queryCompleter.ts]: This section of code should be enhanced so it added a todo there.

**Pull request closed without merging**